etcdserver/api/v3rpc/lease.go

```diff
@@ -86,7 +86,7 @@ func (ls *LeaseServer) LeaseKeepAlive(stream pb.Lease_LeaseKeepAliveServer) erro
 		}
 
 		if err != nil {
-			return err
+			return handleLeaseEOF(err)
```

Comment:
Other RPCs should be handled by gRPC, I think. These only need for RPCs with raw HTTP requests.